nexus/types/src/external_api/params.rs

```diff
@@ -1034,6 +1038,12 @@ pub struct InstanceCreate {
     pub start: bool,
 }
 
+/// Parameters of an `Instance` that can be reconfigured after creation.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct InstanceUpdate {
+    pub boot_disk: Option<NameOrId>,
```

Comment:
it means "unset the boot disk", yeah. i was going to ask if there's a good way to communicate this through the OpenAPI document, which made me realize the new block in `nexus.json` is wrong; it doesn't mention taking a request body at all. i'll fix that and add the also-missing `InstanceUpdate` schema with appropriate docs (and make sure it matches rustdoc here)

i'd thought about trying an `enum FieldUpdate<T> { Set(T), Ignore }` but it really does make client requests kind of awful to write, and it'd be a third mechanism to set fields different from the other ways to update an object. i think it could be good to do, but i'd want to do it in a more considered way than just for instance updating. (importantly, for later readers: some fields _can't_ be "unset", like "number of CPUs", so if there is optionality it has to be in the inner type)

Reply:
We tend to land on 

1. every PUT should include all fields so that `None` can indeed mean unset
2. there are existing PUT endpoints that don't work that way, i.e., use options to let the user leave things untouched

Given that we are already in an inconsistent state, I think we should just build any new things the right way (none means unset) and hope we can fix the wrong ones eventually.

Reply:
You could put a doc comment on that field saying something to the effect of "None will unset the boot disk"

Reply:
done w/ https://github.com/oxidecomputer/omicron/pull/6585/commits/996687ed0a95fde770ba24a5c997241e8a02b2c5